modules/custom/az_event/js/az_calendar_filter.es6.js

```diff
@@ -0,0 +1,205 @@
+/**
+ * @file
+ * A JavaScript file for the datepicker calendar functionality.
+ *
+ */
+
+(($, Drupal, drupalSettings) => {
+  Drupal.behaviors.azCalendarFilter = {
+    attach(context, settings) {
+      const filterInformation = drupalSettings.azCalendarFilter;
+      if (!drupalSettings.hasOwnProperty("calendarFilterRanges")) {
+        drupalSettings.calendarFilterRanges = [];
+      }
+
+      // Drupal settings get merged rather than replaced during ajax.
+      // We should clear out stale entries when we process a new cells.
+      drupalSettings.azCalendarFilter = {};
+
+      // Process cell date strings into javascript dates.
+      for (const property in filterInformation) {
+        if (filterInformation.hasOwnProperty(property)) {
+          drupalSettings.calendarFilterRanges[property] = [];
+          const ranges = filterInformation[property];
+          for (let i = 0; i < ranges.length; i++) {
+            drupalSettings.calendarFilterRanges[property].push([
+              $.datepicker.parseDate("@", ranges[i][0] * 1000),
+              $.datepicker.parseDate("@", ranges[i][1] * 1000),
+            ]);
+          }
+        }
+      }
+
+      // We may have recieved new cell data. Refresh existing datepickers.
+      $(".az-calendar-filter-calendar").datepicker("refresh");
+
+      // Initialize calendar widget wrapper if needed.
```

Comment:
Here is an example of es6 jQuery each in case it helps.  I know that one stumped me for a while. https://github.com/az-digital/az_quickstart/blob/main/modules/custom/az_paragraphs/az_paragraphs_text_media/js/az_paragraphs_az_text_media_youtube.es6.js#L23-L34